experimental/packages/api-events/src/types/EventEmitterProvider.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { EventEmitter } from './EventEmitter';
+import { EventEmitterOptions } from './EventEmitterOptions';
+
+/**
+ * A registry for creating named {@link EventEmitter}s.
+ */
+export interface EventEmitterProvider {
+  /**
+   * Returns an EventEmitter, creating one if one with the given name, version, and
+   * schemaUrl pair is not already created.
+   *
+   * @param name The name of the event emitter or instrumentation library.
+   * @param version The version of the event emitter or instrumentation library.
+   * @param options The options of the event emitter or instrumentation library.
+   * @returns EventEmitter An event emitter with the given name and version
+   */
+  getEventEmitter(
+    name: string,
+    version?: string,
+    options?: EventEmitterOptions
+  ): EventEmitter;
```

Comment:
@martinkuba `domain` is mandatory in events. how do you intend to ensure that?  Would this method fail if `eventDomain` is not present in `EventEmitterOptions` passed? If so, can you make that explicit in the comments documentation? 

Also, have you considered having it as an explicit parameter to this method `getEventEmitter` so that it's not missed by accident?

Reply:
I think that makes sense, and I can make that change. I did not want to make any changes to the API in this PR, so that it can be merged quickly in order to unblock the Log SDK work. But I suppose this is simple enough.